openwisp_monitoring/device/models.py

```diff
@@ -74,6 +74,11 @@ def data_user_friendly(self):
             # convert to GHz
             if 'wireless' in interface and 'frequency' in interface['wireless']:
                 interface['wireless']['frequency'] /= 1000
+            # TODO: Refactor this to a suitable form in which address can be retrieved
+            if 'addresses' in interface:
+                data['addresses'] = {}
+                for count, address in enumerate(interface['addresses']):
+                    data['addresses']['address' + str(count)] = address['address']
```

Comment:
unless I'm missing something, I would keep this data structure as is and just display it as a table in the device status tab.